src/Workspaces/CSharp/Portable/Formatting/Engine/Trivia/TriviaDataFactory.CodeShapeAnalyzer.cs

```diff
@@ -300,6 +297,17 @@ private bool ShouldFormat()
                 return false;
             }
 
+            private bool OnDisabledTextTrivia(SyntaxTrivia trivia, int index)
+            {
+                if (trivia.IsKind(SyntaxKind.DisabledTextTrivia) &&
+                    SyntaxFacts.IsNewLine(trivia.ToString().Last()))
```

Comment:
Last() is an extension method which gets the last char. I should introduce a NullEmpty check.